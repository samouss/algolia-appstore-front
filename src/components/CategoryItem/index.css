.CategoryItem__Label {
  display: flex;
  justify-content: space-between;
  align-items: center;

  &:hover > span {
    color: #0080ff;
  }
}

.CategoryItem__Label--active > span {
  color: #0080ff;
}

.CategoryItem__Stat {
  color: #b9b9b9;
  font-size: 0.8rem;
}
