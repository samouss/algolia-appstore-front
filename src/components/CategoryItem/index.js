import React from 'react';
import PropTypes from 'prop-types';
import { CheckboxInput } from 'components/Input';
import style from './index.css';

export const FacetValuePropTypes = {
  name: PropTypes.string.isRequired,
  count: PropTypes.number.isRequired,
  isRefined: PropTypes.bool.isRequired,
  onChange: PropTypes.func.isRequired,
};

const CategoryItem = ({ name, count, isRefined, onChange }) => (
  <li>
    <CheckboxInput
      customClassName={style.CategoryItem__Label}
      customActiveClassName={style['CategoryItem__Label--active']}
      id={`category-${name}`}
      value={name}
      checked={isRefined}
      onChange={() => onChange(
        name,
        isRefined,
      )}
    >
      {name}
      <span styleName="CategoryItem__Stat">({count})</span>
    </CheckboxInput>
  </li>
);

CategoryItem.propTypes = FacetValuePropTypes;

export default CategoryItem;
