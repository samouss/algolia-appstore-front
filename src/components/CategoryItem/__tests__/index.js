import React from 'react';
import { shallow } from 'enzyme';
import CategoryItem from '../index';

describe('<CategoryItem />', () => {
  const defaultProps = {
    name: 'Games',
    count: 1235,
    isRefined: false,
    onChange: () => {},
  };

  it('expect to render', () => {
    const props = {
      ...defaultProps,
    };

    const component = shallow(
      <CategoryItem
        {...props}
      />,
    );

    expect(component).toMatchSnapshot();
  });

  it('expect to render when refined', () => {
    const props = {
      ...defaultProps,
      isRefined: true,
    };

    const component = shallow(
      <CategoryItem
        {...props}
      />,
    );

    expect(component).toMatchSnapshot();
  });

  it('expect to call onChange', () => {
    const props = {
      ...defaultProps,
      onChange: jest.fn(),
    };

    const component = shallow(
      <CategoryItem
        {...props}
      />,
    );

    component
      .find('CheckboxInput')
      .simulate('change');

    expect(props.onChange).toHaveBeenCalled();
  });
});
