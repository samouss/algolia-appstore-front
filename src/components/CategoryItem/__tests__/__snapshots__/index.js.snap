// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CategoryItem /> expect to render 1`] = `
<li>
  <CheckboxInput
    checked={false}
    customActiveClassName="CategoryItem__Label--active"
    customClassName="CategoryItem__Label"
    id="category-Games"
    onChange={[Function]}
    value="Games"
  >
    Games
    <span
      styleName="CategoryItem__Stat"
    >
      (
      1235
      )
    </span>
  </CheckboxInput>
</li>
`;

exports[`<CategoryItem /> expect to render when refined 1`] = `
<li>
  <CheckboxInput
    checked={true}
    customActiveClassName="CategoryItem__Label--active"
    customClassName="CategoryItem__Label"
    id="category-Games"
    onChange={[Function]}
    value="Games"
  >
    Games
    <span
      styleName="CategoryItem__Stat"
    >
      (
      1235
      )
    </span>
  </CheckboxInput>
</li>
`;
